smack-integration-test/src/main/java/org/igniterealtime/smack/inttest/SmackIntegrationTestFramework.java

```diff
@@ -149,6 +163,22 @@ public static void main(String[] args) throws IOException, KeyManagementExceptio
         System.exit(exitStatus);
     }
 
+    private static String getSpecificationReference(Method method) {
+        final SmackIntegrationTest test = method.getAnnotation(SmackIntegrationTest.class);
+        final SpecificationReference spec = method.getDeclaringClass().getAnnotation(SpecificationReference.class);
+        if (spec == null || spec.document().isBlank()) {
+            return null;
+        }
+        String line = spec.document().trim();
+        if (test.section() != null && !test.section().isBlank()) {
```

Comment:
As far as I can tell, this `null`-check and the one below are not necessary because the annotation type element values can not be `null`. Personally, I would drop those checks, but I leave that up to you.